LTTng/CMakeLists.txt

```diff
@@ -14,20 +14,5 @@ find_package(lttng-ust-tracepoint REQUIRED)
 # Include source
 add_subdirectory(src)
 
-# Only include testing stuff if we are the top level and someone hasn't disabled them explicitly
-option(TRACELOGGING_BUILD_TESTS "Set to OFF to disable building of tests" ON)
-
-if (TRACELOGGING_BUILD_TESTS)
-    if (${CMAKE_PROJECT_NAME} STREQUAL ${PROJECT_NAME})
-        enable_testing()
-
-        if (EXISTS ${PROJECT_SOURCE_DIR}/../external/Catch2/CMakeLists.txt)
-            list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/../external/Catch2/contrib")
-            add_subdirectory("${PROJECT_SOURCE_DIR}/../external/Catch2" Catch2)
-        else ()
-            find_package(Catch2 REQUIRED)
-        endif ()
-
-        add_subdirectory(test)
-    endif ()
-endif ()
+# Use the standard BUILD_TESTING option to disable
+add_subdirectory(test)
```

Comment:
I think you need to do
```
if(BUILD_TESTING)
    add_subdirectory(test)
endif()
```

Reply:
We're always going to build the test subdir, there's no reason not to. BUILD_TESTING controls whether CTest calls the enable_testing() macro to actually hook up the tests into the makefile.

Reply:
It is desirable to allow users to not even build the tests. If the test has dependencies that the user doesn't want to figure out how to provide, the user shouldn't have to install them just to get the project to build. And if the test has some build failure like a C++17 dependency, they user should be able to disable building to skip the issue.